niljs/src/utils/transaction.ts

```diff
@@ -0,0 +1,70 @@
+import type { PublicClient } from "../clients/PublicClient.js";
+import { bytesToHex } from "../encoding/fromBytes.js";
+import type { Hex } from "../types/Hex.js";
+import type { ProcessedReceipt } from "../types/IReceipt.js";
+import { waitTillCompleted } from "./receipt.js";
+
+/**
+ * Represents a transaction that can be awaited for completion
+ */
+export class TransactionResponse {
```

Comment:
It's better to be named Transaction

Reply:
Or maybe better TransactionWaiter. It's better to have more infromation but leave it for future things